### ISSUE TITLE
Authentication errors in V2.0.0

### ISSUE THREAD

**Issue by dlpzx** [awslabs/aws-dataall#687]
### Describe the bug

When upgrading or deploying V2.0 branch `main-v2`, it is not possible to move from the log in window.
No matter if we refresh the page or clean the caches, we still land in the following page.
<img width="1792" alt="image" src="https://github.com/awslabs/aws-dataall/assets/71252798/37a49b9e-105e-49d1-b6e4-a7f66d0527a5">

### How to Reproduce

```
*P.S. Please do not attach files as it's considered a security risk. Add code snippets directly in the message body as much as possible.*
```

### OS

na

### Python version

na

### AWS data.all version

V2.0.0

**Comment by dlpzx** [comment]
@nikpodsh Added some logging in one deployment. Thanks to the logs we can confirm that the first error actually happens when we reload the page in `const user = await Auth.currentAuthenticatedUser();` in `frontend/src/authentication/contexts/CognitoAuthContext.js:77` , there is no `user`

When later we then try to login, `user` is undefined --->` user.email`.... fail with `TypeError`

So we need to investigate what has changed in https://aws-amplify.github.io/amplify-js/api/classes/authclass.html#currentauthenticateduser and in our code

**Comment by anmolsgandhi** [comment]
Closing the issue as it has been resolved successfully by #689. Please feel free to reopen if you have any questions!
